### PR TITLE
4.0

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

File: Admin/CRUD/Context/BatchElementContext.php [hunk 1]
```diff
@@ -13,35 +13,31 @@
 
 use FSi\Bundle\AdminBundle\Admin\Context\ContextAbstract;
 use FSi\Bundle\AdminBundle\Admin\Context\Request\HandlerInterface;
-use FSi\Bundle\AdminBundle\Admin\CRUD\BatchElement;
+use FSi\Bundle\AdminBundle\Admin\CRUD\BatchElement as BatchElementAlias;
```
Comment (szymach) [inline Admin/CRUD/Context/BatchElementContext.php:16]: Why the alias?

File: Admin/CRUD/Context/Request/DataGridBindDataHandler.php [hunk 2]
```diff
@@ -14,48 +14,69 @@
...
+    ) {
+        parent::__construct($eventDispatcher);
+
+        $this->dataGridFormHandler = $dataGridFormHandler;
+    }
+
     public function handleRequest(AdminEvent $event, Request $request): ?Response
     {
-        $event = $this->validateEvent($event);
+        $listEvent = $this->validateEvent($event);
```
Comment (szymach) [inline Admin/CRUD/Context/Request/DataGridBindDataHandler.php:42]: I don't a think method named `validateEvent` should return anything. Either it's name or function should be changed.

File: Behat/Context/FiltersContext.php [hunk 3]
```diff
@@ -25,13 +25,13 @@
 
 class FiltersContext extends AbstractContext
 {
-    private DataSourceFactory $dataSourceFactory;
+    //private DataSourceFactory $dataSourceFactory;
```
Comment (szymach) [inline Behat/Context/FiltersContext.php:28]: Remove?

File: Event/AdminEvents.php [hunk 4]
```diff
@@ -13,5 +13,5 @@
 
 final class AdminEvents
```
Comment (szymach) [inline Event/AdminEvents.php:14]: I think we might as well remove these enum classes and use events' classes constants directly.

File: EventListener/LocaleMenuListener.php [hunk 5]
```diff
@@ -101,6 +99,16 @@ private function getLanguageName(?string $locale = null): string
 
     private function getCurrentLocale(): string
     {
-        return $this->requestStack->getMasterRequest()->getLocale();
+        return $this->getMasterRequest()->getLocale();
+    }
+
+    private function getMasterRequest(): Request
+    {
+        $request = $this->requestStack->getMasterRequest();
```
Comment (szymach) [inline EventListener/LocaleMenuListener.php:107]: Can we use `CurrentRequest` here?

File: features/fixtures/project/src/FixturesBundle/Entity/News.php [hunk 6]
```diff
@@ -57,13 +56,11 @@ class News
 
     /**
      * @ORM\Column(length=255, nullable=true)
```
Comment (szymach) [inline features/fixtures/project/src/FixturesBundle/Entity/News.php:58]: Can we get rid of annotation mappings? You can convert them via console command.

File: features/fixtures/project/src/FixturesBundle/Resources/config/services.xml [hunk 7]
```diff
@@ -89,13 +89,27 @@
             </argument>
             <tag name="admin.element" />
         </service>
+<!--
```
Comment (szymach) [inline features/fixtures/project/src/FixturesBundle/Resources/config/services.xml:92]: Remove?

**Comment by chives** [comment]
@fsi-open

File: Admin/CRUD/Context/Request/DataGridBindDataHandler.php (hunk 2 above)
Comment (szymach) [reply Admin/CRUD/Context/Request/DataGridBindDataHandler.php:42]: What about this one?

Comment (chives) [reply Admin/CRUD/Context/Request/DataGridBindDataHandler.php:42]: I'm open for suggestions

Comment (szymach) [reply Admin/CRUD/Context/Request/DataGridBindDataHandler.php:42]: Just replace this assignment with the condition from the `validateEvent`.

**Comment by chives** [comment]
@fsi-open

**Review by szymach — APPROVED** [review]

**Pull request merged**